lib/set-delegates.js

```diff
@@ -35,16 +35,15 @@ module.exports = function(browserWindow, panel, webview, options) {
     ThemeObserverClass = new ObjCClass({
       utils: null,
 
-      'observeValueForKeyPath:ofObject:change:context:': function() {
+      'observeValueForKeyPath:ofObject:change:context:': function(keyPath,object,change) {
+        const newAppearance = change[NSKeyValueChangeNewKey]
+        const isDark = String(newAppearance.bestMatchFromAppearancesWithNames(['NSAppearanceNameAqua', 'NSAppearanceNameDarkAqua'])) === 'NSAppearanceNameDarkAqua'
```

Comment:
So I think we want to get the theme value of Sketch, not of the system (you can override the theme value in the sketch preferences).

So maybe we need to check if `MSTheme.sharedTheme` is set to "auto" (or whatever is the value to respect the system theme), and if so, get the value from `newAppearance`.

Reply:
I really don't know much about `newAppearance` vs `MSTheme` but this solution seems to work when switching sketch preferences (See screen recording). I also tested switching the OS appearance and it works as expected.

What am I missing? 

![Skpm theme fix](https://user-images.githubusercontent.com/20438532/91700092-ff9add00-eb7d-11ea-9c37-80ae2f445f36.gif)

Reply:
Hum you're probably right. Haven't touch it for a while - I thought the listener would get the value of the system, but if it gets the value of Sketch, all good!